### ISSUE TITLE
Error: Running LUA method 'update' - toolManager.lua:2017: attempt to perform arithmetic on field 'sprayerFillLevel' (a nil value)

### ISSUE THREAD

**Issue by VoltronV** [Courseplay/courseplay#5392]
1st off: **_Thank you to everyone, past and current, volunteering time and effort to work on this project._**
2nd: I only personally only know enough to be dangerous. Good news is I can pretty well understand log files, keeping log clean, solving basic issues (some lua, mainly xml).
3rd only to people new to Courseplay (CP), github: Keeping records of "current mod conflicts" and solved issues has already helped me the past few days clean up some errors I've been having. So THANK YOU! I have solved all issues, bar one, I'm currently having by spending several hours going through tracked issues. Whomever reads this, please spend some time and effort yourself going through issues to attempt to find your problem. If nothing else, just purely out of respect of the time and effort for the people working on this project and having to respond to new issues.

The issue I'm having:
With certain implements, using "Tool Work" (I think Mode 4, but I could be wrong), the course auto generates great. When starting worker, the game hangs meaning that I still have mouse inputs to the cursor, but the camera quits following the tractor & implement combo. The tractor & implement keep moving, but without changing speed or course. Then pressing "stop worker", control resumes as normal. So not a complete crash of the game. Current issue occurs when using both of these mods:
FS19_Lizard2500
FS19_PRT_16
Lizard is a slurry spreader. PRT is a manure spreader. I am getting ready to troubleshoot now by using the game default equipment. If that is successful, I will create a custom mod to the default equipment to replicate a larger piece of equipment in order to test proper function handling.
I will do my best to stay active here and especially monitor this post. Thank you!
[log.txt](https://github.com/Courseplay/courseplay/files/4623416/log.txt)

**Comment by VoltronV** [comment]
Here is the exact spreader as being used right now. I am using game default tractor: The biggest Fendt that does NOT have tracts.
[FS19_Lizard2500.zip](https://github.com/Courseplay/courseplay/files/4623433/FS19_Lizard2500.zip)

**Comment by VoltronV** [comment]
Update: Game default slurry spread works fine.

My theory is issue with the two mods (and many others I've tried functioning the same). So I'm guessing they're ports from FS17; missing the CP-expected "sprayer Fill Level" values? I'll dig through code and see if I can help or solve my own issue with the mods, but leave a solution for others to follow.

**Comment by Tensuko** [comment]
So it is a Mod Issue....

**Comment by VoltronV** [comment]
Ok, I'll double check if it's modhub or not. I might have gone looking elsewhere because slurry or manure spreaders were not on modhub last I looked.

**Comment by VoltronV** [comment]
I'll check a ModHub one now

**Comment by VoltronV** [comment]
Confirmed MOD ISSUE.
I understand now why you only support ModHub items! Sorry for wasting your time ya'll.
TLDR: If you find this by google CP hang freeze , likely a mod issue. only use ModHub mods.

**Comment by VoltronV** [comment]
I did find and confirmed both of these mods were on modhub. I'm reporting the incompatibility (missing value) to the mod creators.

**Comment by Tensuko** [comment]
Can we take this as solved for CP (mod issue) ?
Just to keep our Issues clean, it is getting rly messy right now.

**Comment by Miketeg** [comment]
Got the same toolmanager error yesterday while creating a new course with the Ingame pickup 2014 + an in game trailer on Nebraska Land.

FS19_CoursePlay6/toolManager.lua:193: attempt to compare number with nil

Had to force quit the game, nothing was responding.

Using the latest version.
